var/spack/repos/builtin/packages/py-pysam/package.py

```diff
@@ -26,11 +26,9 @@ class PyPysam(PythonPackage):
     depends_on("py-cython@0.21:", when="@0.14:", type="build")
     depends_on("py-cython@0.17:", type="build")
     depends_on("curl")
-    depends_on("bcftools")
-    depends_on("htslib")
-    depends_on("samtools")
-
-    depends_on("htslib@:1.6", when="@:0.13")
+    depends_on("xz")
 
     def setup_build_environment(self, env):
         env.set("LDFLAGS", self.spec["curl"].libs.search_flags)
+        if self.spec.platform == "darwin":
+            env.remove_flags("LDSHARED", "-bundle")
```

Comment:
Where is this env var being set? Should we just unset it entirely if it's coming from outside of Spack?

Reply:
I have no idea lol. The issue I was running into was that when building `pysam` on macOS, I would get the error `clang: error: invalid argument '-bundle' not allowed with '-dynamiclib'`. I dug into the source code, and it appears that [this modification wasn't happening in spack for some reason](https://github.com/pysam-developers/pysam/blob/v0.19.1/linker_tests/link_pre_489/cy_build.py#L15). It built fine for me outside of spack. This fix is just me manually adding that same modification.

Attached is the build log and env if you're curious.

[spack-build-env.txt](https://github.com/spack/spack/files/9993425/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/9993426/spack-build-out.txt)

Reply:
Prob worth reporting this as a bug to the devs and getting to the bottom of this. I try to avoid including hacks and patches unless the devs are aware of the situation and working on a fix.